### ISSUE TITLE
Null for Non-Nullable Argument

### ISSUE THREAD

**Issue by bhsmither** [cubecart/v6#2985]
For PHP 8.1:
See:
```
https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation
```
(I like this site!)

Found so far, `database.class.php`, near lines 873 and 926, allows the $subject to be _null_ in `preg_match()`. This is deprecated.

**Comment by bhsmither** [comment]
Also, same file, near lines 890 and 949, allows the argument to be _null_ in `strtoupper()`.

**Comment by bhsmither** [comment]
Also `ACP->addTabControl()` near line 144, having passed in _null_.

**Comment by bhsmither** [comment]
Also, products.coupons.inc.php, near line 183,
```
$manufacturer_assigned = array_flip(unserialize($coupon[0]['manufacturer_id']));
```
If a manufacturer was not chosen, then a _null_ will be serialized at line 55, and inserted into the database as `N;`.

When unserialized, the _null_ becomes the argument to `array_flip()`. This is a Fatal Error.

**Comment by abrookbanks** [comment]
This seems to be fixed in #2708 (unless I missed one)
